src/tiled_dataset.py

```diff
@@ -33,20 +31,16 @@ def __init__(
         Return:
             ml_data:        tuple, (data_tensor, mask_tensor)
         """
-        self.data_tiled_uri = data_tiled_uri
-        self.data_client = from_uri(data_tiled_uri, api_key=data_tiled_api_key)
-        self.mask_tiled_uri = mask_tiled_uri
-        if mask_tiled_uri:
-            self.mask_client_one_up = from_uri(
-                mask_tiled_uri, api_key=mask_tiled_api_key
-            )
-            self.mask_client = self.mask_client_one_up["mask"]
-            self.mask_idx = [
-                int(idx) for idx in self.mask_client_one_up.metadata["mask_idx"]
-            ]
+
+        self.data_client = data_tiled_client
+        self.mask_client = None
+        if mask_tiled_client:
+            self.mask_client = mask_tiled_client
```

Comment:
This would break the test as it currently is. Can you look at collections we setup in test_train.py and decide whether we want `mask` or `masks`?

Reply:
> This would break the test as it currently is. Can you look at collections we setup in test_train.py and decide whether we want `mask` or `masks`?

Based on previous discussions with @Wiebke , I believe we want `mask` here. The `mask_tiled_client` here is referring to the `uuid` container which contains metadata of the saved annotation (mask_idx ...), and within this container the actual ArrayClient should have the name of `mask`.

I double checked on the segmentation app side, and seems it is still the case: https://github.com/mlexchange/mlex_highres_segmentation/blob/main/utils/data_utils.py#L203-L235

I will address corresponding changes for the `test_train.py` in a separate comment to keep the thread clear.